k8s/tools/deploy-celery-gke.sh

```diff
@@ -0,0 +1,245 @@
+else
+  echo "--no-cluster specified. Authenticating to pre-existing cluster $CLUSTER_NAME"
+fi
+
+# Authenticate to cluster
+gcloud -q --project $DEVSHELL_PROJECT_ID container clusters get-credentials $CLUSTER_NAME --zone $ZONE
+# Create Kubernetes service account
+kubectl get serviceaccounts $SA_NAME || kubectl create serviceaccount $SA_NAME --namespace default
+gcloud iam service-accounts add-iam-policy-binding $SA_NAME@$DEVSHELL_PROJECT_ID.iam.gserviceaccount.com --role roles/iam.workloadIdentityUser --member "serviceAccount:$DEVSHELL_PROJECT_ID.svc.id.goog[default/$SA_NAME]"
+kubectl annotate serviceaccount $SA_NAME --overwrite --namespace default iam.gke.io/gcp-service-account=$SA_NAME@$DEVSHELL_PROJECT_ID.iam.gserviceaccount.com
+
+# Go to deployment folder to make changes files
+cd $DEPLOYMENT_FOLDER
+
+# Add service account to deployments
+sed -i -e "s/serviceAccountName: .*/serviceAccountName: $SA_NAME/g" turbinia-server.yaml turbinia-worker.yaml redis-server.yaml
+
+# Disable some jobs
+echo "Updating $TURBINIA_CONFIG with disabled jobs"
+sed -i -e "s/^DISABLED_JOBS = .*$/DISABLED_JOBS = $DISABLED_JOBS/g" $TURBINIA_CONFIG
+
+# Update Turbinia config with project info
+echo "Updating $TURBINIA_CONFIG config with project info"
+sed -i -e "s/^INSTANCE_ID = .*$/INSTANCE_ID = '$INSTANCE_ID'/g" $TURBINIA_CONFIG
+sed -i -e "s/^TURBINIA_PROJECT = .*$/TURBINIA_PROJECT = '$DEVSHELL_PROJECT_ID'/g" $TURBINIA_CONFIG
+sed -i -e "s/^TURBINIA_ZONE = .*$/TURBINIA_ZONE = '$ZONE'/g" $TURBINIA_CONFIG
+sed -i -e "s/^TURBINIA_REGION = .*$/TURBINIA_REGION = '$REGION'/g" $TURBINIA_CONFIG
+sed -i -e "s/^CLOUD_PROVIDER = .*$/CLOUD_PROVIDER = 'GCP'/g" $TURBINIA_CONFIG
+
+# Create File Store instance and update deployment files with created instance
+if [[ "$*" != *--no-filestore* ]] ; then  
+  echo "Enabling GCP Filestore API"
+  gcloud -q --project $DEVSHELL_PROJECT_ID services enable file.googleapis.com
+  echo "Creating Filestore instance $FILESTORE_NAME with capacity $FILESTORE_CAPACITY"
+  gcloud -q --project $DEVSHELL_PROJECT_ID filestore instances create $FILESTORE_NAME --file-share=name=$FILESTORE_NAME,capacity=$FILESTORE_CAPACITY --zone=$ZONE --network=name=$VPC_NETWORK
+else
+  echo "Using pre existing Filestore instance $FILESTORE_NAME with capacity $FILESTORE_CAPACITY"
+fi
+
+echo "Updating $TURBINIA_CONFIG config with Filestore configuration and setting output directories"
+FILESTORE_IP=$(gcloud -q --project $DEVSHELL_PROJECT_ID filestore instances describe $FILESTORE_NAME --zone=$ZONE --format='value(networks.ipAddresses)' --flatten="networks[].ipAddresses[]")
+FILESTORE_LOGS="'\/mnt\/$FILESTORE_NAME\/logs'"
+FILESTORE_OUTPUT="'\/mnt\/$FILESTORE_NAME\/output'"
+sed -i -e "s/<IP_ADDRESS>/$FILESTORE_IP/g" turbinia-volume-filestore.yaml
+sed -i -e "s/turbiniavolume/$FILESTORE_NAME/g" *.yaml
```

Comment:
This and the next line are pretty open ended regexes in terms of what they are looking for.  Could we add an anchor or something else to reduce the chance of changing things we don't intend to?

Reply:
IMO my intention was to try and have less anchors as possible where possible so that the k8s files can be as bare bone as possible (not requiring a shell script to deploy them in other forms). The closest I got was just to have an <IP_ADDRESS> anchor.

I did go ahead and specific specific `yaml` files instead though and is executing in context of that directory. I can add another anchor named <TURBINIA_STORAGE> however if you'd like it more specific.

Reply:
Ok, I guess having the file limits is a little better.  As discussed out of band, maybe we can move to something with templates like helm or other in the future.